internal/messagejournal/messagejournal.go

```diff
@@ -0,0 +1,216 @@
+package messagejournal
+
+import (
+	"database/sql"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/redhatinsights/yggdrasil"
+	"github.com/redhatinsights/yggdrasil/ipc"
+)
+
+const messageJournalTemporaryDropString string = `
+	DROP TABLE IF EXISTS runtime;
+`
+const messageJournalTemporaryCreateString string = `
+	CREATE TABLE runtime (
+		id INTEGER NOT NULL PRIMARY KEY,
+		message_id VARCHAR(36) NOT NULL,
+		sent DATETIME NOT NULL,
+		worker_name VARCHAR(128) NOT NULL,
+		response_to VARCHAR(36),
+		worker_event INTEGER,
+		worker_message TEXT
+	);
+`
+const messageJournalPersistentCreateString string = `
+	CREATE TABLE IF NOT EXISTS persistent (
+		id INTEGER NOT NULL PRIMARY KEY,
+		message_id VARCHAR(36) NOT NULL,
+		sent DATETIME NOT NULL,
+		worker_name VARCHAR(128) NOT NULL,
+		response_to VARCHAR(36),
+		worker_event INTEGER,
+		worker_message TEXT
+	);
+`
+
+// MessageJournal is a data structure representing the collection
+// of message journal entries received from worker emitted events and messages.
+type MessageJournal struct {
+	database *sql.DB
+}
+
+type MessageJournalFilter struct {
```

Comment:
Missing doc-string. BTW: My IDE complains that the name starts with package name and it suggests to rename it only to `Filter`.